portafly/src/tests/components/util/AlertContext.test.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { AlertsProvider, useAlertsContext } from 'components'
+import { render } from 'tests/custom-render'
+import { waitFor } from '@testing-library/dom'
+
+const AlertComponent = () => {
+  const { addAlert } = useAlertsContext()
+  React.useEffect(() => addAlert({ id: 'id', title: 'title', variant: 'danger' }), [])
+  return <></>
+}
+
+it('shows an alert that disappears after 8 seconds', async () => {
+  jest.setTimeout(8001)
+  const { getByText, queryByText } = render(
+    <AlertsProvider>
+      <AlertComponent />
+    </AlertsProvider>
+  )
+
+  await waitFor(() => expect(getByText('title')).toBeInTheDocument(), { timeout: 7999 })
```

Comment:
This means the test will be idle for 8 seconds, right? If so, maybe we should look for an alternative with time travel / time freezing / etc

Reply:
Well I don't know exactly why but it doesn't wait for it. The test takes about 50ms. The reason why I put await here is for jest to not complain about having multiple async calls.

When putting a second `await` (in the next line), then the test does take 8 seconds.

Reply:
We could address this in the future with a library like timecop in ruby